variable/test/linalg_test.cpp

```diff
@@ -45,11 +47,10 @@ class LinalgMatrixTest : public ::testing::Test {
 };
 
 TEST_F(LinalgMatrixTest, range_check) {
-  for (scipp::index i : {-1, 0, 1, 2, 3})
-    for (scipp::index j : {-1, 0, 1, 2, 3})
-      if (i < 0 || i > 2 || j < 0 || j > 2)
-        EXPECT_THROW_DISCARD(matrices.elements<Eigen::Matrix3d>(i, j),
-                             std::out_of_range);
-      else
-        EXPECT_NO_THROW_DISCARD(matrices.elements<Eigen::Matrix3d>(i, j));
+  for (const std::string &key :
+       {"xx", "xy", "xz", "yx", "yy", "yz", "zx", "zy", "zz"})
+    EXPECT_NO_THROW_DISCARD(matrices.elements<Eigen::Matrix3d>(key));
+  for (const std::string &key : {"x", "y", "z", "XX"})
+    EXPECT_THROW_DISCARD(matrices.elements<Eigen::Matrix3d>(key),
+                         std::out_of_range);
```

Comment:
Shouldn't the element access tests be in `variable_structure_test.cpp`?

Reply:
I think I had split this up, testing "generic" parts in `variable_structure_test.cpp`, but testing more specifically for Eigen types here.